warehouse/legacy/api/json.py

```diff
@@ -75,18 +77,24 @@ def json_project(project, request):
     return json_release(release, request)
 
 
+@view_config(
+    route_name="legacy.api.json.project_slash",
+    context=Project,
+    decorator=_CACHE_DECORATOR,
+)
+def json_project_slash(project, request):
+    return HTTPMovedPermanently(
+        # Respond with redirect to url without trailing slash
+        request.current_route_path(name=project.name)[:-1],
```

Comment:
I think this should probably be something like this instead:

```python
request.route_path("legacy.api.json.project", name=project.name)
```

Reply:
Good call